### ISSUE TITLE
Open other normalization function 

### ISSUE THREAD

**Issue by GuillaumeErhard** [qubvel-org/segmentation_models.pytorch#983]
Current implementation of Conv2dReLU, which is widely used in the codebase, has the use_batchnorm parameter that was set up first as a boolean before evolving for the use case for inplace batchnorm with an implicit string usage.

Why not modify the current API and use a normalization string parameter, which could be used to set up groupnorm or layernorm, which perform better with smaller batch sizes?

I can do the code quickly for that, and I will also add a warning that use_batchnorm might be deprecated in the future.

**Comment by qubvel** [comment]
Hey, it might be a cool feature! The only thing we have to consider is how to provide additional settings for layernorm or groupnorm. So it might be a dictionary instead, similar to aux_params. For example:

```python
use_norm={"type": "layernorm", "eps": 1e-5}
```
